File: apps/api/app/schemas/workload.py
from pydantic import BaseModel


class WorkloadBucketOut(BaseModel):
    owner_role: str
    assigned_to_id: str | None
    assigned_to_name: str | None
    open_items: int
    blocked_items: int
    escalated_items: int


class WorkloadSummaryOut(BaseModel):
    data: list[WorkloadBucketOut]
    total_open_items: int
    unassigned_items: int

File: packages/shared/src/types/workload.ts
import type { UUID } from './common';

export interface WorkloadBucket {
  owner_role: string;
  assigned_to_id: UUID | null;
  assigned_to_name: string | null;
  open_items: number;
  blocked_items: number;
  escalated_items: number;
  source_linked_tasks: number;
  urgent_tasks: number;
}

export interface WorkloadSummary {
  data: WorkloadBucket[];
  total_open_items: number;
  unassigned_items: number;
  source_linked_tasks: number;
  urgent_tasks: number;
}
